### PR TITLE
fix(graphql-server): Conditionally enable OTel plugin and OTel plugin updates

### LINKED ISSUE #8332: [Bug?]: Experimental (?) `useRedwoodOpenTelemetry` is unconditionally enabled without opt-in as of Redwood 5.x
### What's not working?

The `useRedwoodOpenTelemetry` plugin -- which currently seems to be the foundation for an experimental feature (#7829) -- seems to be hardwired into `createGraphQLHandler`, and there doesn't appear to be a way to disable or configure it.

image

We are already using a customized OpenTelemetry plugin (`based on @envelop/opentelemetry`, but customized for our needs) in our app that does similar work, and `useRedwoodOpenTelemetry` is causing essentially duplicate event traffic for us.

It's possible we could switch to using `useRedwoodOpenTelemetry` instead of our custom plugin once the former can be configured for our needs -- primarily setting options like `{ resolvers: false, result: false, variables: true }` to reduce noise, and starting an **active** span -- but in the meantime we would appreciate a way to opt out of `useRedwoodOpenTelemetry`.

It seems as though `useRedwoodOpenTelemetry` shouldn't (yet?) be enabled in the first place without opting in to the Experimental RedwoodJS Studio feature referenced above, but I'm not certain of the intent since I can't find any other references to it.

### How do we reproduce the bug?

_No response_

### What's your environment? (If it applies)

_No response_

### Are you interested in working on this?

- [ ] I'm interested in working on this

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**Problem**
1. The redwood OTel plugin was being enabled automatically rather than being opt-in. This causes issues for users who want to use a different OpenTelemetry setup.
2. It was not possible to pass options to the OTel plugin.

**Changes**
1. The redwood OTel plugin is now only enabled when the `openTelemetryOptions` property is passed in by the user where they create their graphql server/handler.
2. The `openTelemetryOptions` provides a mechanism to provide the 3 options that the existing envelop plugin has. (see https://github.com/n1ru4l/envelop/tree/main/packages/plugins/opentelemetry) 
3. The OTel plugin now starts active spans using `startActiveSpan`
4. The setup command now requires the user to add this `openTelemetryOptions` property manually and so the messaging has been updated. (I will also update the forum post)

**Fixes**
Fixes #8332 

**Notes**
@pvenable - This is long overdue so sorry about that! I have included the three standard options via this new `openTelemetryOptions` property which you would use as shown below. I also used `startActiveSpan` and I _believe_ this would suit your needs although I have no test project with the particular setup you mentioned. 

I would be very, very grateful if you could try this out when it makes it into a release and feedback again on your experience. It would be good if we could make redwood do what you need out of the box!

```ts 
export const handler = createGraphQLHandler({
  authDecoder,
  getCurrentUser,
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect()
  },
  openTelemetryOptions: {
    resolvers: true,
    result: true,
    variables: true,
  }
})
```

@dthyresson - This did not change the behaviour we have before where the results of `makeMergedSchema` were being reported directly under the query span rather than under any particular resolver. I continue to question this but I think we were happy with it before.

### CODE REVIEW

File: packages/graphql-server/src/types.ts [hunk 1]
```diff
@@ -75,6 +75,12 @@ export interface RedwoodGraphQLContext {
   [index: string]: unknown
 }
 
+export interface RedwoodOpenTelemetryConfig {
```
Comment (dthyresson) [inline packages/graphql-server/src/types.ts:78]: Could we perhaps have some inline comments to describe what enabling each does?

for example, setting resolvers to true adds ... ?

This?

```
          if (options.resolvers) {
             extendContext({
               [tracingSpanSymbol]: executionSpan,
```

Adds execution span telemetry for each graph resolver?

**Review by dthyresson — CHANGES_REQUESTED** [review]
I think some inline docs/comments as to how to enable each Otel setting might be helpful.

Also, do these https://redwoodjs.com/docs/graphql#graphql-handler-setup docs need updating for the config options?

File: packages/graphql-server/src/types.ts (hunk 1 above)
Comment (Josh-Walker-GM) [reply packages/graphql-server/src/types.ts:78]: Added the documentation to the options

**Comment by Josh-Walker-GM** [comment]
I think we shouldn't update the docs for the docs website until this moves out of experimental. I will update the forum post.

**Comment by pvenable** [comment]
Thanks, @Josh-Walker-GM, it's great to see this!  I look forward to testing it and providing further feedback once it's released.

I can tell you that since our previous correspondence I've also needed to add some custom attributes to our active span (such as the current user ID or other current context), so I would likely be interested in seeing a way to augment the options/attributes passed to `startActiveSpan` here.  This could perhaps be something like a callback that would run in `onExecute` that could return additional variables to set in the span options, but that's just one idea.

Thanks again for the followup. 🙂

**Pull request merged**